libbeat/otelbeat/otelmap/otelmap.go

```diff
@@ -32,191 +33,53 @@ func ToMapstr(m pcommon.Map) mapstr.M {
-		case []int8:
-			dest := out.PutEmptySlice(k)
-			for _, i := range v.([]int8) {
-				newVal := dest.AppendEmpty()
-				newVal.SetInt(int64(i))
-			}
-		case int16:
-			out.PutInt(k, int64(v.(int16)))
-		case []int16:
-			dest := out.PutEmptySlice(k)
-			for _, i := range v.([]int16) {
-				newVal := dest.AppendEmpty()
-				newVal.SetInt(int64(i))
-			}
-		case int32:
-			out.PutInt(k, int64(v.(int32)))
-		case []int32:
-			dest := out.PutEmptySlice(k)
-			for _, i := range v.([]int32) {
-				newVal := dest.AppendEmpty()
-				newVal.SetInt(int64(i))
-			}
-		case int64:
-			out.PutInt(k, v.(int64))
-		case []int64:
-			dest := out.PutEmptySlice(k)
-			for _, i := range v.([]int64) {
-				newVal := dest.AppendEmpty()
-				newVal.SetInt(i)
-			}
-		case uint:
-			out.PutInt(k, int64(v.(uint)))
-		case []uint:
-			dest := out.PutEmptySlice(k)
-			for _, i := range v.([]uint) {
-				newVal := dest.AppendEmpty()
-				newVal.SetInt(int64(i))
-			}
-		case uint8:
-			out.PutInt(k, int64(v.(uint8)))
-		case []uint8:
-			dest := out.PutEmptySlice(k)
-			for _, i := range v.([]uint8) {
-				newVal := dest.AppendEmpty()
-				newVal.SetInt(int64(i))
-			}
-		case uint16:
-			out.PutInt(k, int64(v.(uint16)))
-		case []uint16:
-			dest := out.PutEmptySlice(k)
-			for _, i := range v.([]uint16) {
-				newVal := dest.AppendEmpty()
-				newVal.SetInt(int64(i))
-			}
-		case uint32:
-			out.PutInt(k, int64(v.(uint32)))
-		case []uint32:
-			dest := out.PutEmptySlice(k)
-			for _, i := range v.([]uint32) {
-				newVal := dest.AppendEmpty()
-				newVal.SetInt(int64(i))
-			}
-		case uint64:
-			out.PutInt(k, int64(v.(uint64)))
-		case []uint64:
-			dest := out.PutEmptySlice(k)
-			for _, i := range v.([]uint64) {
-				newVal := dest.AppendEmpty()
-				newVal.SetInt(int64(i))
-			}
-		case float32:
-			out.PutDouble(k, float64(v.(float32)))
-		case []float32:
-			dest := out.PutEmptySlice(k)
-			for _, i := range v.([]float32) {
-				newVal := dest.AppendEmpty()
-				newVal.SetDouble(float64(i))
-			}
-		case float64:
-			out.PutDouble(k, v.(float64))
-		case []float64:
-			dest := out.PutEmptySlice(k)
-			for _, i := range v.([]float64) {
-				newVal := dest.AppendEmpty()
-				newVal.SetDouble(i)
-			}
-		case bool:
-			out.PutBool(k, x)
-		case []bool:
-			dest := out.PutEmptySlice(k)
-			for _, i := range v.([]bool) {
-				newVal := dest.AppendEmpty()
-				newVal.SetBool(i)
-			}
+// ConvertNonPrimitive handles the conversion of non-primitive data types to pcommon-primitive types.
+// The conversion is performed in place.
+// Notes:
+//  1. Slices require special handling when converting a map[string]any to pcommon.Map.
+//     The pcommon.Map expects all slices to be of type []any.
+//     If you attempt to use other slice types (e.g., []string or []int),
+//     pcommon.Map.FromRaw(...) will return an "invalid type" error.
+//     To overcome this, we use "reflect" to transform []T into []any.
+func ConvertNonPrimitive(m mapstr.M) {
```

Comment:
This removes the default case where we reported an unknown type.  What do you think about adding a new case for all the [pcommon primitive types that FromRaw supports](https://github.com/open-telemetry/opentelemetry-collector/blob/2526b02cf452965c74d137aa2f518c22a8b19e1f/pdata/pcommon/value.go#L134) which is just a `break`, and then restoring the default case for the unknown type?  The advantage I see is that we would produce the error in the function that would need to be changed to support the new type, otherwise we get the error in the otelconsumer and FromRaw function.  Today that isn't a big deal, but in 2 or 3 years it would make it much easier to spot where we need to make the change.